launcher/src/main/java/org/apache/spark/launcher/CommandBuilderUtils.java

```diff
@@ -314,18 +319,31 @@ static int javaMajorVersion(String javaVersion) {
    * or a distribution directory.
    */
   static String findJarsDir(String sparkHome, String scalaVersion, boolean failIfNotFound) {
+    String envSparkJarsDir = System.getenv(ENV_SPARK_JARS_DIR);
+    if (envSparkJarsDir != null && !envSparkJarsDir.isEmpty()) {
+      File envJarsDir = new File(envSparkJarsDir);
+      return handleJarsDir(envJarsDir, failIfNotFound,
```

Comment:
`return`? What I suggested is to warn and continue the old lookup behavior.
```
SPARK_JARS_DIR -> sparkHome/jars -> assembly/target/scala-%s/jars
```

Reply:
I will fix it. Thank you.